comixed-frontend/src/app/xhr.interceptor.ts

```diff
@@ -77,7 +77,7 @@ export class XhrInterceptor implements HttpInterceptor {
             return;
           case 401:
             this.logger.error('[XHR] user not authenticated:', error);
-            this.authenticationAdaptor.startLogout();
+            this.authenticationAdaptor.authenticationFailed();
```

Comment:
Do you think we could add a boolean field to AuthenticationState, called "authenticating", to indicate whether or not we're in the process of doing a login?

Then here the code could do something like:
```
if(this.authenticationAdaptor.authenticating) {
  this.logger.debug('Authentication attempt faied');
  this.authenticationAdaptor.authenticationFailed();
} else {
  this.logger.debug('No valid auth token');
  this.authenticationAdaptor.startLogout();
}
```

I think this would treat the two scenarios (trying to login vs. trying to access a page with an expired or no authentication token) differently and let us show the right toaster message.

Reply:
Sounds like a good idea indeed.